rust/vectordb/src/data/inspect.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright 2023 Lance Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use arrow::compute::kernels::{aggregate::bool_and, length::length};
+use arrow_array::{cast::AsArray, types::Int32Type, Array, RecordBatchReader};
+use arrow_ord::comparison::eq_dyn_scalar;
+use arrow_schema::DataType;
+
+use crate::error::{Error, Result};
+
+pub(crate) fn infer_dimension(arr: &dyn Array) -> Result<Option<i32>> {
+    if !matches!(arr.data_type(), DataType::List(_)) {
+        return Ok(None);
+    };
+    let list_arr = arr.as_list_opt::<i32>().expect("must be a list now");
+    let len_arr = length(list_arr)?;
+    if len_arr.is_empty() {
+        return Ok(Some(0));
+    }
+
+    let dim = len_arr.as_primitive::<Int32Type>().value(0);
+    if bool_and(&eq_dyn_scalar(len_arr.as_primitive::<Int32Type>(), dim)?) != Some(true) {
+        return Ok(None);
+    }
+    return Ok(Some(dim));
+}
+
+/// Infer the vector columns from a dataset.
+///
+/// Parameters
+/// ----------
+/// - reader: RecordBatchReader
+/// - strict: if set true, only fixed_size_list<float> is considered as vector column. If set to false,
+///           a list<float> column with same length is also considered as vector column.
+pub fn infer_vector_columns(
+    reader: impl RecordBatchReader + Send,
+    strict: bool,
+) -> Result<Vec<String>> {
+    let mut columns = vec![];
+
+    let mut columns_map: HashMap<String, Option<i32>> = HashMap::new();
```

Comment:
nit: should we call this `columns_to_infer`?

Reply:
Done.